x/auction/legacy/v0_17/migrate.go

```diff
@@ -0,0 +1,42 @@
+package v0_17
+
+import (
+	"strings"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+
+	v016auction "github.com/kava-labs/kava/x/auction/legacy/v0_16"
+	v017auction "github.com/kava-labs/kava/x/auction/types"
+)
+
+func migrateAuction(auction *codectypes.Any) *codectypes.Any {
+	auction.TypeUrl = strings.Replace(auction.TypeUrl, "v1beta1", "v1beta2",-1)
```

Comment:
v17 version of the old v16 tests would be good, maybe also one that makes sure the encoded bytes doesn't change between versions.

If the encoded byte data did change, another way could be using unpacking, converting to the new v17 type, then re-packing it as any.

Reply:
In this case the encoded bytes _will_ change since the type url is being updated. I'm still a protobuf novice, do you have time tomorrow to pair on the unpacking / any approach?